### PR TITLE
feat(#141): Created better tests structure

### LINKED ISSUE #141: Create better tests structure
We are starting to get a lot of tests and eventually it is going to be a huge file for `sqlite`. What I propose is that we do something like this instead:

```
tests
    sqlite
        common_test.rs
        insert_test.rs
        update_test.rs
        delete_test.rs        
        select_test.rs
    postgres
        ...
```

### PR DESCRIPTION
**Pull request by Hiccup-za**
GitHub Issue: https://github.com/mjovanc/njord/issues/141

# Changes

- Created the `sqlite` folder & split tests out into separate test files
- Addressed all warnings within test files
- Updated the README with a Testing section

Let me know what you think!

# Testing

All tests:
all tests

All tests within sqlite folder:
all sqlite tests

A specific test within the sqlite folder:
specific test

### CODE REVIEW

File: njord/tests/sqlite/open_test.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+use njord::keys::{AutoIncrementPrimaryKey, PrimaryKey};
+use njord::sqlite::{self};
+use njord::table::Table;
+use njord_derive::Table;
+use std::path::Path;
+
+#[derive(Table, Clone)]
```
Comment (mjovanc) [inline njord/tests/sqlite/open_test.rs]: We dont need the structs here. Also maybe name this to "base" or "core" or something in that manner just so we can have other tests that does not apply in the others.

File: njord/tests/sqlite/update_test.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+use njord::condition::Condition;
+use njord::keys::{AutoIncrementPrimaryKey, PrimaryKey};
+use njord::sqlite::select::SelectQueryBuilder;
+use njord::sqlite::{self};
+use njord::table::Table;
+use njord_derive::Table;
+use std::collections::HashMap;
+use std::path::Path;
+
+#[derive(Table, Clone)]
```
Comment (mjovanc) [inline njord/tests/sqlite/update_test.rs]: Just make sure we dont have any structs here that is not in use. Maybe you could take a look at having these structs in mod.rs instead? So we dont need to define them multiple times. Only have it in the file if it's unique.

**Review by mjovanc — COMMENTED** [review]
This is amazing work! I just left two comments if you can take a look at those and I will merge it.

**Comment by mjovanc** [comment]
Also add this to the contributors section in `README.md`:

`[Hiccup-za](https://github.com/Hiccup-za)`

**Comment by mjovanc** [comment]
Also, add "Testing" to the TOC in `README.md`.

File: njord/tests/sqlite/open_test.rs (hunk 1 above)
Comment (Hiccup-za) [reply njord/tests/sqlite/open_test.rs]: Actioned ✅

File: njord/tests/sqlite/update_test.rs (hunk 2 above)
Comment (Hiccup-za) [reply njord/tests/sqlite/update_test.rs]: Actioned ✅

**Comment by mjovanc** [comment]
LGTM!

**Pull request merged**